### DISCUSSION TITLE
Timeout in Modbus Server RTU

### DISCUSSION THREAD

**Discussion by engineerarslan** [eModbus/eModbus#150] [Q&A] [answered]
What is the meaning of timeout in ModbusServerRTU.cpp line number 16?

ModbusServerRTU::ModbusServerRTU(HardwareSerial& serial, uint32_t timeout, int rtsPin)

I am using this library to create a modbus server/slave and I want to light up an error led in case of no communication with the master. i.e If the server does not hear from the master for certain amount of time then an alarm should light up. Can this timeout feature be used for that?

**Comment by Miq1** [comment, marked as answer]
No, not really. With the exception of the coming Modbus ASCII mode with RTU servers the timeout has no use yet, since the servers are supposed to listen all the time regardless.

If you need to check for regular requests from a client, you could use a millis() based timestamp that is reset every time a request is processed in the server's callback functions you provide and will trigger an action in loop() if a certain amount of time has passed.
